src/services/expenseServices.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+let expenses = [];
+
+function resetExpenses() {
+  expenses = [];
+}
+
+function getExpenses(userId, categories, from, to) {
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => Number(expense.userId) === Number(userId));
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => categories.includes(expense.category));
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => (
+        expense.spentAt > from
+        && expense.spentAt < to
+      ));
+  }
+
+  if (from && !to) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.spentAt >= from);
+  }
+
+  if (to && !from) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.spentAt <= to);
+  }
+
+  return filteredExpenses;
+};
+
+function getExpenseByUserId(userId) {
+  const foundExpense = expenses
+    .find(expense => expense.userId === Number(userId));
+
+  return foundExpense;
+}
+
+function getExpenseById(id) {
+  const foundExpense = expenses
+    .find(expense => expense.id === Number(id));
+
+  return foundExpense;
+}
+
+function createExpense({
+  userId,
+  spentAt,
+  title,
+  amount,
+  category,
+  note,
+}) {
+  const newExpense = {
+    id: expenses.length + 1,
```

Comment:
Make a separate function to `getMaxId` or `getNextId` & reuse it in all cases instead of taking a length + 1/
If i have 2 items & delete first item, after that try to add one more - i will have 2 items with the same id

Reply:
or use `uuid` lib